src/libs/tmdk/e2e-tests/basic-e2e/basic-e2e.test.ts

```diff
@@ -0,0 +1,492 @@
+  const model1S3Location =
+    "s3://" + workspaceS3BucketName + "/" + constants.model1FileName;
+  const model2S3Location =
+    "s3://" + workspaceS3BucketName + "/" + constants.model2FileName;
+
+  // 1. Clean up pre-existing resources, if any
+  console.log(
+    "Deleting test IAM role, s3 bucket, and workspace if they exist."
+  );
+  try {
+    await aws().tm.getWorkspace({ workspaceId: constants.workspaceId });
+    console.log("Workspace exists, nuking it.");
+    argv2 = {
+      _: ["nuke"],
+      $0: "tmdk_local",
+      region: "us-east-1",
+      "workspace-id": constants.workspaceId,
+    } as Arguments<nuke.Options>;
+    await nuke.handler(argv2);
+    await aws().tm.deleteWorkspace({ workspaceId: constants.workspaceId });
+  } catch (e) {
+    console.log("Workspace does not exist, as expected.");
+  }
+  try {
+    await aws().s3.send(
+      new DeleteObjectCommand({
+        Bucket: workspaceS3BucketName,
+        Key: constants.model1FileName,
+      })
+    );
+    await aws().s3.send(
+      new DeleteObjectCommand({
+        Bucket: workspaceS3BucketName,
+        Key: constants.model2FileName,
+      })
+    );
+    await aws().s3.send(
+      new DeleteObjectCommand({
+        Bucket: workspaceS3BucketName,
+        Key: constants.scene1FileName,
+      })
+    );
+    await aws().s3.send(
+      new DeleteObjectCommand({
+        Bucket: workspaceS3BucketName,
+        Key: constants.scene2FileName,
+      })
+    );
+    await aws().s3.deleteBucket({ Bucket: workspaceS3BucketName });
+  } catch (e) {
+    console.log("S3 bucket does not exist, as expected.");
+  }
+  try {
+    await aws().iam.deleteRolePolicy({
+      RoleName: twinMakerRoleName,
+      PolicyName: twinMakerPolicyName,
+    });
+  } catch (e) {
+    console.log("IAM role policy does not exist, as expected.");
+  }
+  try {
+    await aws().iam.deleteRole({ RoleName: twinMakerRoleName });
+  } catch (e) {
+    console.log("IAM role does not exist, as expected.");
+  }
+
+  // 2. Set up test resources
+  try {
+    console.log("Creating IAM role: " + twinMakerRoleName);
+    await aws().iam.createRole({
+      RoleName: twinMakerRoleName,
+      AssumeRolePolicyDocument: JSON.stringify(twinMakerAssumeRolePolicy),
+    });
+    let twinMakerPolicyString = JSON.stringify(twinMakerPermissionPolicy);
+    twinMakerPolicyString = twinMakerPolicyString.replace(
+      "s3ArnStar",
+      workspaceS3BucketArn + "/*"
+    );
+    twinMakerPolicyString = twinMakerPolicyString.replace(
+      "s3ArnStandard",
+      workspaceS3BucketArn
+    );
+    twinMakerPolicyString = twinMakerPolicyString.replace(
+      "s3ArnDelete",
+      workspaceS3BucketArn + "/DO_NOT_DELETE_WORKSPACE_*"
+    );
+    await aws().iam.putRolePolicy({
+      RoleName: twinMakerRoleName,
+      PolicyName: twinMakerPolicyName,
+      PolicyDocument: twinMakerPolicyString,
+    });
+    await delay(10000); // allow role to propagate
```

Comment:
can we avoid the blind wait? (or make TODO to revisit)

Reply:
Sure, will add TODO to verify role creation